Assets/Scripts/DataTypes/Vertices/CrossSection.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace DataTypes
+{
+    public class CrossSection : Vertex
+    {
+        protected Anchor upAnchor;
+        protected Anchor rightAnchor;
+        protected Anchor downAnchor;
+        protected Anchor leftAnchor;
+        
+        public CrossSection(IEnumerable<Anchor> anchors) : base(anchors)
+        {
+            if (this.anchors.Length != 4)
+            {
+                throw new WrongAnchorCount("Cross sections require exactly 4 anchors!");
+            }
+
+            this.upAnchor = this.anchors[0];
+            this.rightAnchor = this.anchors[1];
+            this.downAnchor = this.anchors[2];
+            this.leftAnchor = this.anchors[3];
+        }
+
+        public CrossSection(Anchor upAnchor, Anchor rightAnchor, Anchor downAnchor, Anchor leftAnchor)
+            : base(ImmutableArray.Create(upAnchor, rightAnchor, downAnchor, leftAnchor))
```

Comment:
Man könnte auch den Constructor von CrossSection selbst callen, also statt base this.

Reply:
76029d2b25d11d6a2e8ce509161b72ffe3495fbe & 22bc498ed7eb7018bf84fc792b922c7400f4f3b8